### ISSUE TITLE
gdal_grid: Allow catching 'failed to open datasource', not just skipping

### ISSUE THREAD

**Issue by jidanni** [OSGeo/gdal#9630]
### Feature description

```
gdal_grid -a linear -outsize 8 8 -l nnn nnn.vrt nnn.tiff; echo RET: $?
ERROR 1: Failed to open datasource `nnn.csv'.
No point geometry found on layer nnn, skipping.
RET: 0
```

### Additional context

There should be a way to have RET be 1, not 0.

Else, well, the rocket might leave the launch pad without the astronauts aboard.

**Comment by rouault** [comment]
This is actually a duplicate of #9406 that has been fixed in 3.9.0dev / master

**Comment by jidanni** [comment]
Here's what's now going on:
```
$ gdal_grid -l 0 -zfield field_13 CSV:0.locations 0.tif
Failed to find field field_13 on layer 0, skipping.
$ echo $?
1
$ gdal_grid --version
GDAL 3.9.0, released 2024/05/07
```
It is good that it now returns 1, and not 0.
But what still needs to be done is:
1. Say "ERROR..." (because you set $?), don't just say "skipping" anymore.
2. Do not create the half-baked output file and leave it sitting on disk.


Also shouldn't this cause an error too?
```
gdal_grid -l 0 -zfield field_3 CSV:0.locations 0.tif
No point geometry found on layer 0, skipping.
```
